### ISSUE TITLE
Back to top component

### ISSUE THREAD

**Issue by matthewcarleton** [patternfly/patternfly-design#970]
For pages that have a lot of content, it would be nice to drop in a back to top component.
In OCP there are quite a few pages that could use this.

**Comment by mcarrano** [comment]
@matthewcarleton do you have any designs for this? Couldn't this be accomplished by just adding a button or link in the page that navigates back to the h1 heading at the top of the page? Unclear why we would need a new component.

**Comment by matthewcarleton** [comment]
We don't have a design for it. It did strike me as something that PF could deliver because we could have a consistent icon as well as any scrolling animation that would be applied. You could just have a simple "Back to the top" anchor that takes you back to the top but it would be nice to have a more polished/consistent element across the different products.

**Comment by mceledonia** [comment]
I think this will be relatively simple from a visual perspective, some kind of pill-shape button anchored to bottom center or right and floating on top of the page content.

A big consideration here is covering important information at the bottom of the page. Therefore it would be good to consider how the user would remove the button if necessary. Sometimes this can be covered by tying the appearance of the button to scroll behavior - scroll down and it appears, scroll up (even a little bit) and it goes away. The other idea that comes to mind is having a dismiss action, or some combination of the two.

Happy to take a look at this!

**Comment by matthewcarleton** [comment]
ya that's a good point @mceledonia 
I think it would be very useful to have in OCP for sure. There are many long lists there :)

**Comment by mcarrano** [comment]
@matthewcarleton we will get this on our roadmap. Is this needed in any particular timeframe?

**Comment by beanh66** [comment]
Good idea @matthewcarleton, might be a POC-worthy thing to just vet out in the console and contribute back possibly!

**Comment by coreyvickery** [comment]
@mceledonia Just for reference, this is what I created for our design system website. Obviously a big edge case, but just sharing because my team might need something official soon.

https://ux.redhat.com/

**Comment by coreyvickery** [comment]
@mcarrano Forgot to tag you, see above.

**Comment by mcarrano** [comment]
Proposed designs from @bgliwa01 

![back to top.png](https://images.zenhubusercontent.com/5ce816524e6c594dc1b0df4d/f9709480-8da1-4b0b-be4e-732f188b51ed)

**Comment by mcarrano** [comment]
@bgliwa01 Here is the feedback from the PatternFly design share meeting on June 23:

* Prefer full text version (“Back to top”) from a content perspective.
* Prefer side by side  + 16px spacing (bottom right option). Have we considered icon after text vs before? It also might make sense to try reducing the font size.
* The component should include a drop shadow to indicate that it is floating over the background.
* Would be good to see a mockup of this placed in context.

Let's discuss next steps.

**Comment by seanforyou23** [comment]
Was wondering, this component seems to have an awful lot in common with [SkipToContent](https://www.patternfly.org/v4/components/skip-to-content) and JumpLinksItem from the [JumpLinks](https://www.patternfly.org/v4/components/jump-links) component. Other than the styling/positioning and maybe animation, what is the difference in terms of behavior for what BackToTop should do?

Just wanted to ask in case it makes more sense to combine this as a feature of one of those existing components. Feel free to ignore me if it sounds like a bad idea :)

**Comment by mcarrano** [comment]
@mcoker thoughts about the above?

**Comment by bgliwa01** [comment]
![back to top](https://user-images.githubusercontent.com/51342070/127220429-de657fe2-a122-4a9e-9667-e6419a425ac4.png)

**Comment by mcoker** [comment]
@seanforyou23 it seems like the main overlap is that they're both a primary button as a link that takes you to an anchor on the page, though the skip to content component uses the primary button as it is and this component deviates from the default styling in a number of ways - padding, border-radius, font-size, and box-shadow. Like you mentioned there will also be positioning, and likely animation differences.

One question I ask if something should be a new component or a variation of an existing one is, would I expect most changes to the component to apply to the base component and the variation? I wouldn't think so in this case (at least as far as the styling is concerned). Also the naming (skip to content), while we can update that later, doesn't really match.

Is there a benefit of them being the same from a react perspective @seanforyou23?

**Comment by seanforyou23** [comment]
Makes sense @mcoker, there could be some benefit but after hearing the other considerations (especially the naming) I think you're probably right - not worth the effort. Thx for entertaining the question!

**Comment by bgliwa01** [comment]
![Default - mobile xs](https://user-images.githubusercontent.com/51342070/127696016-014d540c-09c2-4452-a7e6-73fc7206c31d.png)
![backtotoppage](https://user-images.githubusercontent.com/51342070/127695970-eb2e40fc-3e92-407f-8cbf-5f8ac112d3a5.png)

**Comment by mcarrano** [comment]
Thanks @bgliwa01 . I have opened a new core issue here to begin implementation: https://github.com/patternfly/patternfly/issues/4255
